executor/builder.go

```diff
@@ -260,51 +278,79 @@ func (b *executorBuilder) buildIndexScan(v *plan.IndexScan) Executor {
 		} else {
 			ex = b.buildUnionScanExec(e)
 		}
+		ex = b.buildFilter(ex, v.AccessConditions)
 		return b.buildFilter(ex, remained)
 	}
 
-	var idx table.Index
-	for _, val := range tbl.Indices() {
-		if val.Meta().Name.L == v.Index.Name.L {
-			idx = val
-			break
-		}
-	}
 	e := &IndexScanExec{
 		tbl:         tbl,
 		tableAsName: v.TableAsName,
 		idx:         idx,
 		fields:      v.Fields(),
 		ctx:         b.ctx,
 		Desc:        v.Desc,
-		valueTypes:  make([]*types.FieldType, len(idx.Meta().Columns)),
-	}
-
-	for i, ic := range idx.Meta().Columns {
-		col := tbl.Cols()[ic.Offset]
-		e.valueTypes[i] = &col.FieldType
+		valueTypes:  valueTypes,
 	}
 
 	e.Ranges = make([]*IndexRangeExec, len(v.Ranges))
 	for i, val := range v.Ranges {
-		e.Ranges[i] = b.buildIndexRange(e, val)
+		e.Ranges[i], _ = b.buildIndexRange(e, val, idx.Meta().Columns)
 	}
-	x := b.buildFilter(e, v.FilterConditions)
+	x := b.buildFilter(e, v.AccessConditions)
+	x = b.buildFilter(x, v.FilterConditions)
```

Comment:
If prefix index is used, we can just build one filter by `append(v.AccessConditions, v.FilterConditions...)`